components/automate-ui/src/app/pages/notification-details/notification-details.component.html

```diff
@@ -0,0 +1,103 @@
+          <chef-option value='details'>Details</chef-option>
+        </chef-tab-selector>
+      </chef-page-header>
+      <section class="page-body">
+        <form [formGroup]="updateForm">
+          <chef-form-field>
+            <label for="name">Name <span aria-hidden="true">*</span></label>
+            <input chefInput id="name" formControlName="name" type="text" [resetOrigin]="saveSuccessful" autocomplete="off" data-cy="name-input"/>
+            <chef-error *ngIf="(nameCtrl.hasError('required') || nameCtrl.hasError('pattern')) && nameCtrl.dirty">
+              Name is required.
+            </chef-error>
+          </chef-form-field>
+          <chef-form-field>
+            <label>
+              <span class="label">Webhook type<span aria-hidden="true">*</span></span>
+              <chef-select
+                (change)="changeSelectionForWebhookType($event)"
+                class="notification-select-menu"
+                [value]="notification.targetType">
+                <chef-option *ngFor="let key of getTagetTypeKeys()" [value]="key">
+                  <img src="assets/img/logo-{{key}}.png" alt="{{key}} logo" />
+                  {{ notification.TargetTypeLabels[key] }}
+                </chef-option>
+              </chef-select>
+            </label>
+          </chef-form-field>
+          
+          <chef-form-field>
+            <label>
+              <span class="label">Failure type  <span aria-hidden="true">*</span></span>
+              <chef-select
+                (change)="setFailureType($event)"
+                class="notification-select-menu"
+                [value]="notification.ruleType">
+                <chef-option *ngFor="let key of getAlertTypeKeys()" [value]="key">
+                  {{ notification.AlertTypeLabels[key] }}
+                </chef-option>
+              </chef-select>
+              <div *ngIf="displayCriticalControlsCheckbox()">
+                <div class="small">
+                  <chef-checkbox
+                    (change)="updateCriticalControlsOnly($event.detail)"
+                    [checked]="notification.criticalControlsOnly">
+                  </chef-checkbox>
+                  Only send notifications for critical control failures
+                </div>
+              </div>
+            </label>
+          </chef-form-field>
+          <chef-form-field>
+            <label for="targetUrl">Webhook URL <span aria-hidden="true">*</span></label>
+            <input chefInput id="targetUrl" formControlName="targetUrl" type="text" [resetOrigin]="saveSuccessful" autocomplete="off" data-cy="targetUrl-input"/>
+            <chef-error *ngIf="(urlCtrl.hasError('required') || urlCtrl.hasError('pattern')) && urlCtrl.dirty">
+              Webhook URL is required.
+            </chef-error>
+          </chef-form-field>
+        </form>
+        <div>
+          <chef-button secondary [disabled]="testInProgress || !urlPresent()" (click)="sendTestForTargetUrl()">
```

Comment:
Unfortunately we really shouldn't use functions in the template.  In this case, `!urlPresent()` - have a look at the console in the video posted below.  You'll see that the `urlPresent()` function runs continuously.

![continuous function](https://user-images.githubusercontent.com/16737484/84923817-8f031a80-b07c-11ea-950f-0b213e590b40.gif)

There is more than one way to handle this, however @msorens used a pure pipe recently in this example which only fires when a change is happens: https://github.com/chef/automate/pull/3876#discussion_r435590088